resteasy-client/src/main/java/org/jboss/resteasy/client/jaxrs/engines/ApacheHttpAsyncClient4Engine.java

```diff
@@ -190,6 +194,49 @@ public ClientResponse extractResult(ClientResponse response)
       }
    }
 
+   @Override
+   public <T> CompletableFuture<T> submit(ClientInvocation request,
+         boolean buffered,
+         ResultExtractor<T> extractor,
+         ExecutorService executorService) {
+      if (buffered) {
+         final CompletableFuture<T> cf = new CompletableFuture<>();
+         final InvocationCallback<T> callback = new InvocationCallback<T>()
```

Comment:
This would cause `Response` object to be closed.  So, any chaining on `CompletableFuture` will not be able to work on `Response` object (especially if it is happening on a separate thread).  We can use `FutureCallback`  without a `InvocationCallback` in the picture, actually.

Reply:
Please see Javadoc if `InvocationCallback`.  Sorry, I have to go offline now :(